Clarify rule notification values

lcawl opened the pull request:
## Summary

Fixes https://github.com/elastic/kibana/issues/137270

In particular, this PR:

1. Fixes the hover help for the `Notify` option to pertain to alert statuses instead of rule statues.
2. Clarifies the text strings for the `Notify` options.
3. Clarifies the `schedule`, `throttle`, and `notify_when` properties in the create rule API
4. Removes the details about the `Notify` options from the [Create and manage rules](https://www.elastic.co/guide/en/kibana/current/create-and-manage-rules.html) page, since it's more accurately covered in the UI and in the API.
5. Adds alert status definitions in [Create and manage rules](https://www.elastic.co/guide/en/kibana/current/create-and-manage-rules.html) alongside the rule status definitions, since they don't appear in any of the reference information yet.

## Screenshots

### Before

![image](https://user-images.githubusercontent.com/26471269/189457622-05daadbc-ef03-4a4f-a29c-30a6434e423a.png)

![image](https://user-images.githubusercontent.com/26471269/189457665-ddfc44be-0c91-4d27-9e18-cea2dbc03341.png)


### After

![image](https://user-images.githubusercontent.com/26471269/189755142-82607c84-3a41-4cbb-86ec-d8e14a2cea1b.png)


![image](https://user-images.githubusercontent.com/26471269/190184305-89baf13e-042d-43de-a335-abfd3257bf3e.png)

lcawl commented on x-pack/plugins/triggers_actions_ui/public/application/sections/rule_form/rule_notify_when.tsx:
```diff
@@ -77,7 +77,7 @@ export const NOTIFY_WHEN_OPTIONS: Array<EuiSuperSelectOption<RuleNotifyWhenType>
           <p>
             <FormattedMessage
-              defaultMessage="Actions repeat at the rule interval when the alert is active."
+              defaultMessage="Actions repeat at each check interval."
```
Alternatively, could have this option as:

**Every check interval**
Actions repeat if the alert is active.

ymao1 replied:
A wordier explanation if this helps: `Actions run when alert becomes active and repeat at each check interval if alert remains active`.

lcawl replied:
Thanks, I've updated the docs and the UI text to reflect that info.

lcawl commented on docs/user/alerting/create-and-manage-rules.asciidoc:
```diff
@@ -197,6 +188,11 @@ A rule can have one of the following statuses:
 `pending`:: The rule has not yet run.  The rule was either just created, or enabled after being disabled.
 `unknown`:: A problem occurred when calculating the status. Most likely, something went wrong with the alerting code.
 
+Each alert also has one of the following statuses:
+
+`active`:: The conditions for the rule are met, and actions should be generated according to the notification settings.
+`recovered`:: The conditions for the rule are no longer met.
+
```
I added this section since I couldn't find anywhere else we take about alert statuses. However, if those are indeed the only valid status options, I'm confused about when/if the "only on status change" option would ever fire. i.e. If the notify options pertain only to alerts are "active" and yet we're telling it to only repeat actions when the status changes, does that mean they only repeat when an alert has gone from "recovered" to "active"?

ymao1 replied:
Only on status change would only fire when alert goes from active to recovered or recovered to active. Some rules have multiple action groups, for example `error` and `warn` so both `error` and `warn` would be considered `active` but actions would fire if moving from `error` to `warn` or `warn` to `recovered` or `recovered` to `error`, for example.

lcawl replied:
Thanks!  I have updated the text to (1) remove "while the alert is active" from the "Notify when" hover, and (2) updated the alert status docs to clarify that "active" and "recovered" are just two examples. If you think we ought to cover the other possibilities there as well, let me know.

elasticmachine commented:
Pinging @elastic/response-ops (Team:ResponseOps)

ymao1 commented on x-pack/plugins/triggers_actions_ui/public/application/sections/rule_form/rule_notify_when.tsx:
```diff
@@ -49,7 +49,7 @@ export const NOTIFY_WHEN_OPTIONS: Array<EuiSuperSelectOption<RuleNotifyWhenType>
           <p>
             <FormattedMessage
-              defaultMessage="Actions run when the alert status changes."
+              defaultMessage="Actions repeat if the alert status changes."
```
I don't know if `repeat` is accurate here. As written above, an action would be triggered when moving between statuses, but I wouldn't consider this repeating.

lcawl replied:
Thanks, changed!

ersin-erdal commented:
IMHO, "Every time alert is active" is also wrong.
Actions for "recovered alerts" are also triggered when this option is selected. So, not only for active alerts...

lcawl commented on docs/user/alerting/create-and-manage-rules.asciidoc:
```diff
@@ -197,6 +188,11 @@ A rule can have one of the following statuses:
 `pending`:: The rule has not yet run.  The rule was either just created, or enabled after being disabled.
 `unknown`:: A problem occurred when calculating the status. Most likely, something went wrong with the alerting code.
 
+Each alert also has one of the following statuses:
```
```suggestion
Alerts also have statuses, for example:
```

lcawl commented:
> IMHO, "Every time alert is active" is also wrong. Actions for "recovered alerts" are also triggered when this option is selected. So, not only for active alerts...

Thanks, I will follow up with you to understand this better, since it isn't clear to me when alerts would *stop* generating actions in this case.

ymao1 reviewed (approved):
LGTM

lcawl commented:
> IMHO, "Every time alert is active" is also wrong. Actions for "recovered alerts" are also triggered when this option is selected. So, not only for active alerts...

Thanks for the information!  I have changed the menu options so that none of them refer to "active" alerts. I also updated the description of "recovered" alerts to be "The conditions for the rule are no longer met, and recovery actions should be generated". If these or any other changes are not accurate, please let me know.

ersin-erdal commented:
> > IMHO, "Every time alert is active" is also wrong. Actions for "recovered alerts" are also triggered when this option is selected. So, not only for active alerts...
> 
> Thanks for the information! I have changed the menu options so that none of them refer to "active" alerts. I also updated the description of "recovered" alerts to be "The conditions for the rule are no longer met, and recovery actions should be generated". If these or any other changes are not accurate, please let me know.

All LGTM, thank you for pinging me.
